### PR TITLE
Don't require wildcard cert for deployments

### PR DESCRIPTION
**Pull request by schnie**
This is what I hacked in to create a basic POC. Needs obvious cleanup. Just posting to get the ball rolling.

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
fix https://github.com/astronomer/issues/issues/459

**Comment by andriisoldatenko** [comment]
@schnie @ashb PTAL

File: src/routes/v1/authorization/handler.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default async function(req, res) {
 
   // Parse out some variables.
   const originalUrl = req.get("x-original-url");
-  const { hostname } = url.parse(originalUrl);
+  const { hostname, path } = url.parse(originalUrl);
```
Comment (andriisoldatenko) [inline src/routes/v1/authorization/handler.js]: @schnie should i check subdomain `deployments` or permissions here?

Comment (schnie) [reply src/routes/v1/authorization/handler.js]: I think below where we check if we have `matches` we can also check is subdomain is `deployments`

File: src/routes/v1/authorization/handler.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ export default async function(req, res) {
 
     // If we have permission, authorize it.
     if (airflowRoles.length > 0) {
-      const jwt = exports.airflowJWT(user, airflowRoles, hostname);
+      const jwt = exports.airflowJWT(user, airflowRoles, matches[1]);
```
Comment (ashb) [inline src/routes/v1/authorization/handler.js]: I'd feel happier here if we also included the basedomain in the audience too

File: src/routes/v1/authorization/handler.js (hunk 1 above)
Comment (andriisoldatenko) [reply src/routes/v1/authorization/handler.js]: @schnie done.

File: src/routes/v1/authorization/handler.js [hunk 3]
```diff
@@ -28,19 +28,25 @@ export default async function(req, res) {
...
     const deploymentId = await prisma
-      .deployment({ releaseName: matches[1] })
+      .deployment({ releaseName: releaseName })
       .id();
 
     // Check if we have deployment level access to it.
     const airflowRoles = mapLocalRolesToAirflow(user, deploymentId);
 
+    // Prepare audience based on https://tools.ietf.org/html/rfc7519#section-4.1.3
+    const audience = [subdomain, releaseName];
```
Comment (andriisoldatenko) [inline src/routes/v1/authorization/handler.js]: @ashb FYI

File: src/routes/v1/authorization/handler.js [hunk 4]
```diff
@@ -28,19 +28,23 @@ export default async function(req, res) {
...
     const deploymentId = await prisma
-      .deployment({ releaseName: matches[1] })
+      .deployment({ releaseName: releaseName })
       .id();
 
     // Check if we have deployment level access to it.
     const airflowRoles = mapLocalRolesToAirflow(user, deploymentId);
 
+    // Prepare audience based on https://tools.ietf.org/html/rfc7519#section-4.1.3
+    const audience = [hostname, releaseName].join("/");
```
Comment (andriisoldatenko) [inline src/routes/v1/authorization/handler.js]: @schnie ^^^

**Pull request merged**